### PR TITLE
[web] Adding profile parameter to firefox

### PR DESCRIPTION
**Pull request by nturgut**
Firefox sometimes gives a popup related to "corrupt profile" and asks if a new profile should be created (happened on local Mac). This might create a flake in the future on LUCI bots

In order to prevent these type of popups give a profile when opening firefox. Preferences are taken from: https://github.com/dart-lang/test/blob/master/pkgs/test/lib/src/runner/browser/firefox.dart

### CODE REVIEW

File: lib/web_ui/dev/firefox.dart [hunk 1]
```diff
@@ -43,13 +44,23 @@ class Firefox extends Browser {
...
+      final _profile = '''
+user_pref("browser.shell.checkDefaultBrowser", false);
+user_pref("dom.disable_open_during_load", false);
+user_pref("dom.max_script_run_time", 0);
+''';
+
       // A good source of various Firefox Command Line options:
       // https://developer.mozilla.org/en-US/docs/Mozilla/Command_Line_Options#Browser
       //
       var dir = createTempDir();
```
Comment (yjbanov) [inline lib/web_ui/dev/firefox.dart:57]: Where does this temp dir go? We should, as much as possible, use `environment.webUiDartToolDir` for any outputs our tools produce. For example, `.dart_tool/firefox_profile` would be a good and predictable place to put firefox profile data.

File: lib/web_ui/dev/firefox.dart [hunk 2]
```diff
@@ -43,13 +44,23 @@ class Firefox extends Browser {
...
+      File(path.join(dir, 'prefs.js')).writeAsStringSync(_profile);
```
Comment (yjbanov) [inline lib/web_ui/dev/firefox.dart:58]: Should we clean up the profile directory between test runs?

Comment (nturgut) [reply lib/web_ui/dev/firefox.dart:58]: We delete the directory after processExits: 
Line 78: unawaited(process.exitCode
          .then((_) => Directory(dir).deleteSync(recursive: true)));

File: lib/web_ui/dev/firefox.dart (hunk 1 above)
Comment (nturgut) [reply lib/web_ui/dev/firefox.dart]: done.

**Comment by nturgut** [comment]
This fix is related to Firefox, which only runs on Linux Web Engine luci bot. Merging the PR since the test failures are not related to the code path.

**Pull request merged**